### PR TITLE
OBPIH-4930 Show catalog colors on outbound shipment pages

### PR DESCRIPTION
**Pull request by alannadolny**

### CODE REVIEW

**Review by kchelstowski — APPROVED** [review]

File: grails-app/init/org/pih/warehouse/BootStrap.groovy [hunk 1]
```diff
@@ -293,6 +293,7 @@ class BootStrap {
                 "inventoryItem.id"      : picklistItem.inventoryItem?.id,
                 "product.name"          : picklistItem?.inventoryItem?.product?.name,
                 "product.displayName"   : picklistItem?.inventoryItem?.product?.displayName,
+                "product.color"         : picklistItem?.inventoryItem?.product?.color,
```
Comment (jmiranda) [inline grails-app/init/org/pih/warehouse/BootStrap.groovy:296]: I'm cool with this one for now. But eventually, I will replace all of these fake nested objects with a proper JSON object graph. I'm going to be working on these issues over the next few months so let's keep that in mind and let me know if there's anything in particular we'll need to deal with on the client side when that happens.

File: grails-app/views/requisition/_showRequisitionItem.gsp [hunk 2]
```diff
@@ -90,22 +90,28 @@
         <g:elseif test="${requisitionItem?.isSubstituted()}">
             <div class="canceled">
             <g:link controller="inventoryItem" action="showStockCard" id="${requisitionItem?.product?.id}">
-                <format:displayName product="${requisitionItem?.product}" showTooltip="${true}" />
+                <span style="color: ${requisitionItem?.product?.color ?: "black"}">
```
[the remark this answers is not in the document]
Comment (jmiranda) [reply grails-app/views/requisition/_showRequisitionItem.gsp:93]: Lastly, is there any advantage to moving the display logic into the format:displayName tag? In particular, we could standardize on using that tag any time we need to display the product name. However, we'd have to potentially make some changes to the tag to support HTML vs non-HTML callers. In other words, we don't want to add a span tag to JSON object used in a non-HTML context.

**Review by jmiranda — CHANGES_REQUESTED** [review]
One overarching concern would be with performance. I can't remember if "color" property is an N+1 query or a formula. But if it's backed by an N+1 query then we could potentially have a 4*N+1 query problem here. Actually, more since we have at least one forEach loops in there. 

So we should consider moving that to a separate tag (format:displayProductWithColor) that invokes the format:displayProduct tag.

And then we can probably use caching on top of that. 

https://grails.github.io/grails-cache/4.0.3/guide/index.html#cacheTags

File: grails-app/views/requisition/_showRequisitionItem.gsp (hunk 2 above)
[the remark this answers is not in the document]
Comment (alannadolny) [reply grails-app/views/requisition/_showRequisitionItem.gsp:93]: I don't think so, that moving this display logic to this tag is a good idea. In the ticket, it was described to add colors on specific pages - adding colors to this tag will affect more than those listed pages.

**Comment by alannadolny** [comment]
@jmiranda I can see that the color is a formula. I created a custom tag for adding color to the product name, and I also used caching as you suggested.

**Review by awalkowiak — APPROVED** [review]
Looks like all change requests were resolved, and it lgtm.

**Pull request merged**